src/model/errors/R2Error.ts

```diff
@@ -1,9 +1,15 @@
+interface R2ErrorExtraAction {
+    actionFunction: () => void;
```

Comment:
Naming nitpick: if the interface is named "Action", then just "function" would suffice - on the call site is would read "action.function()". If "ButtonProps" is used for the interface, this could be "buttonProps.action()" or "buttonProps.function"().

Also could future proof this by using `Promise<void>` as the return value.